### PR TITLE
feat: allow different bot name to auto-merge

### PR DESCRIPTION
**Pull request by qcastel**
This Pull Request fulfills the following requirements:

- [x] The commit message follows our guidelines.
- [x] Tests for the changes have been added if needed.
- [x] Does not affect documentation or it has been added or updated.

I am trying to contribute an improvement that allows a different bot to auto-merge.
I did try this github action and it works well, now I am trying to contribute back to this repo.

I did adapt the test accordingly and some doc.

Let me know what you think!

### CODE REVIEW

**Comment by merlinnot** [comment]
Thanks for the contribution, makes sense to me! @viestat Can you please review? In this case we might also consider renaming the both 🤔

**Comment by merlinnot** [comment]
@viestat @acazacu On a second thought, since people want to use different bots, maybe it would be wise to accept a RegExp here too, or at least some minimatch?

**Comment by viestat** [comment]
This makes sense to me. There is something that I like to point out;  this change does not tie the auto merge action to a bot name, rather to a github login. Meaning that if I set the `BOT_NAME` value to `viestat` for example, it will auto-merge every pull request I open 😄

This is not a bad thing at all but the focus of the action is different then. Therefore we should rename the action to express what it does and change the variable name used from `BOT_NAME` to `GITHUB_LOGIN`

File: README.md [hunk 1]
```diff
@@ -59,6 +59,21 @@ auto-merge:
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 ```
 
+### Enable auto-merge for a different bot
+
+You may have another bot that also raises PR against your repo and you want to
+auto-merge. By default, this github action will assume the bot is dependabot.
+You can override the bot name by setting up the value `BOT_NAME`:
```
Comment (viestat) [inline README.md]:
```suggestion
You can override the bot name by setting up the value `GITHUB_LOGIN`:
```

File: README.md [hunk 2]
```diff
@@ -59,6 +59,21 @@ auto-merge:
...
+
+```yaml
+steps:
+  - name: Auto merge
+    uses: ridedott/dependabot-auto-merge-action@master
+    with:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      BOT_NAME: my-awesome-bot-r2d2
```
Comment (viestat) [inline README.md]:
```suggestion
      GITHUB_LOGIN: my-awesome-bot-r2d2
```

File: action.yml [hunk 3]
```diff
@@ -7,6 +7,10 @@ inputs:
   GITHUB_TOKEN:
     description: 'A GitHub token.'
     required: true
+  BOT_NAME:
```
Comment (viestat) [inline action.yml]:
```suggestion
  GITHUB_LOGIN:
```

File: action.yml [hunk 4]
```diff
@@ -7,6 +7,10 @@ inputs:
...
+    description: 'The bot name on which you want the auto-merge to be enabled'
```
Comment (viestat) [inline action.yml]: There is no real limitation to not use a `bot` account. So this is the Github login that will have the PRs auto merged

File: src/eventHandlers/checkSuite/index.ts [hunk 5]
```diff
@@ -3,17 +3,19 @@
...
 import { context, GitHub } from '@actions/github';
 
-import { DEPENDABOT_GITHUB_LOGIN } from '../../constants';
 import { findPullRequestInfo } from '../../graphql/queries';
 import { mutationSelector } from '../../util';
 
-export const checkSuiteHandle = async (octokit: GitHub): Promise<void> => {
+export const checkSuiteHandle = async (
+  octokit: GitHub,
+  BOT_NAME: string,
```
Comment (viestat) [inline src/eventHandlers/checkSuite/index.ts]:
```suggestion
  GITHUB_LOGIN: string,
```

File: src/eventHandlers/checkSuite/index.ts [hunk 6]
```diff
@@ -3,17 +3,19 @@
...
+): Promise<void> => {
   const pullRequests = context.payload.check_suite.pull_requests;
 
   for (const pullRequest of pullRequests) {
     if (
       typeof context.payload.sender === 'object' &&
-      context.payload.sender.login === DEPENDABOT_GITHUB_LOGIN
+      context.payload.sender.login === BOT_NAME
```
Comment (viestat) [inline src/eventHandlers/checkSuite/index.ts]:
```suggestion
      context.payload.sender.login === GITHUB_LOGIN
```

File: src/eventHandlers/checkSuite/index.ts [hunk 7]
```diff
@@ -67,7 +69,7 @@ export const checkSuiteHandle = async (octokit: GitHub): Promise<void> => {
         warning(JSON.stringify(error));
       }
     } else {
-      info('Pull request not created by Dependabot, skipping.');
+      info(`Pull request not created by ${BOT_NAME}, skipping.`);
```
Comment (viestat) [inline src/eventHandlers/checkSuite/index.ts]:
```suggestion
      info(`Pull request not created by ${GITHUB_LOGIN}, skipping.`);
```

File: src/eventHandlers/pullRequest/index.ts [hunk 8]
```diff
@@ -2,16 +2,18 @@ import { info, warning } from '@actions/core';
...
 import { PayloadRepository } from '@actions/github/lib/interfaces';
 
-import { DEPENDABOT_GITHUB_LOGIN } from '../../constants';
 import { findPullRequestLastApprovedReview } from '../../graphql/queries';
 import { mutationSelector } from '../../util';
 
-export const pullRequestHandle = async (octokit: GitHub): Promise<void> => {
+export const pullRequestHandle = async (
+  octokit: GitHub,
+  BOT_NAME: string,
```
Comment (viestat) [inline src/eventHandlers/pullRequest/index.ts]:
```suggestion
  GITHUB_LOGIN: string,
```

File: src/eventHandlers/pullRequest/index.ts [hunk 9]
```diff
@@ -2,16 +2,18 @@ import { info, warning } from '@actions/core';
...
+): Promise<void> => {
   const pullRequest = context.payload.pull_request;
 
   if (pullRequest === undefined) {
     warning('Pull request information is unavailable.');
-  } else if (pullRequest.user.login === DEPENDABOT_GITHUB_LOGIN) {
+  } else if (pullRequest.user.login === BOT_NAME) {
```
Comment (viestat) [inline src/eventHandlers/pullRequest/index.ts]:
```suggestion
  } else if (pullRequest.user.login === GITHUB_LOGIN) {
```

File: src/eventHandlers/pullRequest/index.ts [hunk 10]
```diff
@@ -47,6 +49,6 @@ export const pullRequestHandle = async (octokit: GitHub): Promise<void> => {
       warning(JSON.stringify(error));
     }
   } else {
-    info('Pull request not created by Dependabot, skipping.');
+    info(`Pull request not created by ${BOT_NAME}, skipping.`);
```
Comment (viestat) [inline src/eventHandlers/pullRequest/index.ts]:
```suggestion
    info(`Pull request not created by ${GITHUB_LOGIN}, skipping.`);
```

File: src/eventHandlers/push/index.ts [hunk 11]
```diff
@@ -24,8 +23,11 @@ const getReferenceName = (): string => {
   return name;
 };
 
-export const pushHandle = async (octokit: GitHub): Promise<void> => {
-  if (context.payload.pusher.name === DEPENDABOT_GITHUB_LOGIN) {
+export const pushHandle = async (
+  octokit: GitHub,
+  BOT_NAME: string,
```
Comment (viestat) [inline src/eventHandlers/push/index.ts]:
```suggestion
  GITHUB_LOGIN: string,
```

File: src/eventHandlers/push/index.ts [hunk 12]
```diff
@@ -24,8 +23,11 @@ const getReferenceName = (): string => {
...
+): Promise<void> => {
+  if (context.payload.pusher.name === BOT_NAME) {
```
Comment (viestat) [inline src/eventHandlers/push/index.ts]:
```suggestion
  if (context.payload.pusher.name === GITHUB_LOGIN) {
```

File: src/eventHandlers/push/index.ts [hunk 13]
```diff
@@ -75,6 +77,6 @@ export const pushHandle = async (octokit: GitHub): Promise<void> => {
       warning(JSON.stringify(error));
     }
   } else {
-    info('Pull request not created by Dependabot, skipping.');
+    info(`Pull request not created by ${BOT_NAME}, skipping.`);
```
Comment (viestat) [inline src/eventHandlers/push/index.ts]:
```suggestion
    info(`Pull request not created by ${GITHUB_LOGIN}, skipping.`);
```

File: src/index.ts [hunk 14]
```diff
@@ -10,15 +11,18 @@ import {
 const GITHUB_TOKEN = getInput('GITHUB_TOKEN');
 
 const octokit = new GitHub(GITHUB_TOKEN);
+const BOT_NAME = getInput('BOT_NAME');
```
Comment (viestat) [inline src/index.ts]:
```suggestion
const GITHUB_LOGIN = getInput('GITHUB_LOGIN');
```

File: src/index.ts [hunk 15]
```diff
@@ -10,15 +11,18 @@ import {
...
 
 const main = async (): Promise<void> => {
+  info(`The bot name is ${BOT_NAME}`);
```
Comment (viestat) [inline src/index.ts]:
```suggestion
  info(`The bot name is ${GITHUB_LOGIN}`);
```

File: src/index.ts [hunk 16]
```diff
@@ -10,15 +11,18 @@ import {
...
+
   switch (context.eventName) {
     case 'check_suite':
-      return checkSuiteHandle(octokit);
+      return checkSuiteHandle(octokit, BOT_NAME);
```
Comment (viestat) [inline src/index.ts]:
```suggestion
      return checkSuiteHandle(octokit, GITHUB_LOGIN);
```

File: src/index.ts [hunk 17]
```diff
@@ -10,15 +11,18 @@ import {
...
     case 'pull_request':
-      return pullRequestHandle(octokit);
+      return pullRequestHandle(octokit, BOT_NAME);
```
Comment (viestat) [inline src/index.ts]:
```suggestion
      return pullRequestHandle(octokit, GITHUB_LOGIN);
```

File: src/index.ts [hunk 18]
```diff
@@ -10,15 +11,18 @@ import {
...
     case 'push':
-      return pushHandle(octokit);
+      return pushHandle(octokit, BOT_NAME);
```
Comment (viestat) [inline src/index.ts]:
```suggestion
      return pushHandle(octokit, GITHUB_LOGIN);
```

**Comment by viestat** [comment]
Moving to https://github.com/ridedott/dependabot-auto-merge-action/pull/233 

@qcastel we really appreciate your contribution, thank you!

**Pull request closed without merging**